src/utils/addresses.test.ts

```diff
@@ -23,9 +23,27 @@ describe('utils', () => {
     })
   })
 
-  describe('#shortenAddress', () => {
+  describe('#shortenAddressStrict', () => {
     it('throws on invalid address', () => {
-      expect(() => shortenAddress('abc')).toThrow("Invalid 'address'")
+      expect(() => shortenAddressStrict('abc')).toThrow("Invalid 'address'")
+    })
+
+    it('truncates middle characters', () => {
+      expect(shortenAddressStrict('0xf164fc0ec4e93095b804a4795bbe1e041497b92a')).toBe('0xf164...b92a')
+    })
+
+    it('truncates middle characters even without prefix', () => {
+      expect(shortenAddressStrict('f164fc0ec4e93095b804a4795bbe1e041497b92a')).toBe('0xf164...b92a')
+    })
+
+    it('renders checksummed address', () => {
+      expect(shortenAddressStrict('0x2E1b342132A67Ea578e4E3B814bae2107dc254CC'.toLowerCase())).toBe('0x2E1b...54CC')
+    })
+  })
+
+  describe('#shortenAddress', () => {
```

Comment:
add a test for a for an alternate number of chars at the beginning and another for alternate at the end